Antidote/CallViewController.m

```diff
@@ -0,0 +1,184 @@
+
+    UIView *darkView = [[UIView alloc] initWithFrame:self.view.bounds];
+    darkView.backgroundColor = [UIColor blackColor];
+    darkView.alpha = 0.5;
+
+    [self.view addSubview:darkView];
+
+    UIVisualEffect *blurEffect = [UIBlurEffect effectWithStyle:UIBlurEffectStyleLight];
+
+    UIVisualEffectView *visualEffectView = [[UIVisualEffectView alloc] initWithEffect:blurEffect];
+    visualEffectView.frame = self.view.bounds;
+
+    [self.view addSubview:visualEffectView];
+}
+
+- (void)createNameLabel
+{
+    if (self.call.chat.friends.count == 1) {
+        OCTFriend *friend = self.call.chat.friends.firstObject;
+
+        CGRect labelFrame = CGRectMake(0, 0, 200, 200);
+        self.nameLabel = [[UILabel alloc] initWithFrame:labelFrame];
+        self.nameLabel.text = friend.nickname;
+        self.nameLabel.textColor = [UIColor whiteColor];
+        self.nameLabel.textAlignment = NSTextAlignmentCenter;
+        [self.nameLabel sizeToFit];
+
+        [self.view addSubview:self.nameLabel];
+    }
+}
+- (void)createEndCallButton
+{
+    CGRect frame = CGRectMake(0, 0, 400, 100);
+    self.cancelCallButton = [[UIButton alloc] initWithFrame:frame];
+    self.cancelCallButton.backgroundColor = [UIColor redColor];
+    [self.cancelCallButton setImage:[UIImage imageWithContentsOfFile:@"phone"] forState:UIControlStateNormal];
+    [self.cancelCallButton addTarget:self action:@selector(endCall) forControlEvents:UIControlEventTouchUpInside];
+
+    [self.view addSubview:self.cancelCallButton];
+}
+
+- (void)createCallTimer
+{
+    self.timerLabel = [[UILabel alloc] init];
+    self.timerLabel.textColor = [UIColor whiteColor];
+    self.timerLabel.textAlignment = NSTextAlignmentCenter;
+
+    [self.view addSubview:self.timerLabel];
+}
+
+- (void)layoutSubviews
+{
+    [self.nameLabel mas_makeConstraints:^(MASConstraintMaker *make) {
+        make.top.equalTo(self.view.mas_top).with.offset(50);
+        make.left.equalTo(self.view.mas_left).with.offset(50);
+        make.right.equalTo(self.view.mas_right).with.offset(-50);
+        make.height.equalTo(30);
+    }];
+
+    [self.cancelCallButton mas_makeConstraints:^(MASConstraintMaker *make) {
+        make.bottom.equalTo(self.view.mas_bottom).with.offset(-50);
+        make.centerX.equalTo(self.view.mas_centerX);
+        make.left.equalTo(self.view.mas_left).with.offset(50);
+        make.right.equalTo(self.view.mas_right).with.offset(-50);
+    }];
+
+    [self.timerLabel mas_makeConstraints:^(MASConstraintMaker *make) {
+        make.top.equalTo(self.nameLabel.mas_bottom).with.offset(20);
+        make.left.equalTo(self.nameLabel.mas_left);
+        make.right.equalTo(self.nameLabel.mas_right);
+        make.centerX.equalTo(self.nameLabel.centerX);
+    }];
+}
+
+- (NSString *)convertToTimeFromInterval:(NSTimeInterval)interval
+{
+    int minutes = (int)interval / 60;
+    int seconds = interval - (minutes * 60);
+
+    return [NSString stringWithFormat:@"%02d:%02d", minutes, seconds];
```

Comment:
There's stuff for creating NSDate objects but I couldn't find anything for converting time interval to MM:SS. There was NSNumberFormatter and NSDateFormatter, but those don't seem to fit my need.